src/traversals/bfs.jl

```diff
@@ -1,98 +1,19 @@
-### DEVELOPERS NOTE: BFS optimization experiments are typically
-### prototyped in gdistances!, since it's one of the simplest
-### BFS implementations.
+#### REMOVE-2.0
 
-"""
-    tree(parents)
-
-Convert a parents array into a directed graph.
-"""
-function tree(parents::AbstractVector{T}) where T <: Integer
-    n = T(length(parents))
-    t = DiGraph{T}(n)
-    for (v, u) in enumerate(parents)
-        if u > zero(T)  && u != v
-            add_edge!(t, u, v)
-        end
-    end
-    return t
+function bfs_parents(g::AbstractGraph, s::Integer; dir = :out)
+    Base.depwarn("`bfs_parents` is deprecated. Equivalent functionality has been moved to `LightGraphs.Traversals.parents`.", :bfs_parents)
```

Comment:
maybe the tests from <=1.3 can stay with `@test_deprecated`

Reply:
As discussed, this doesn't buy us much and will force us to change existing working "legacy" tests which might pollute things later on, for what will amount to a few-month minor advantage.